### PR TITLE
mobile/ci: Bump macos -> 14

### PR DESCRIPTION
**Pull request by phlax**

### CODE REVIEW

File: mobile/ci/mac_ci_setup.sh [hunk 1]
```diff
@@ -55,7 +55,7 @@ do
 done
 
 # https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md#xcode
-sudo xcode-select --switch /Applications/Xcode_14.1.app
+sudo xcode-select --switch "/Applications/Xcode_${XCODE_VERSION}.app"
```
Comment (BillyAutrey) [inline mobile/ci/mac_ci_setup.sh:58]: I usually have to do it like this when I'm working locally.
```suggestion
sudo xcode-select --switch "/Applications/Xcode_${XCODE_VERSION}.app/Contents/Developer"
```

Comment (phlax) [reply mobile/ci/mac_ci_setup.sh:58]: ill try that - but i think there is still a mismatch in what is setup with `rules_apple` and what is available/being provided

Comment (BillyAutrey) [reply mobile/ci/mac_ci_setup.sh:58]: Disregard, I was looking at the wrong build.

**Review by abeyad — DISMISSED** [review]

**Comment by phlax** [comment]
@abeyad testing this in staging currently - looking good so far, will update when confirmed working

/wait

File: .github/workflows/mobile-ios_build.yml [hunk 2]
```diff
@@ -89,7 +89,7 @@ jobs:
       container-command:
       docker-ipv6: false
       request: ${{ needs.load.outputs.request }}
-      runs-on: macos-12-large
+      runs-on: macos-14-xlarge
```
Comment (abeyad) [inline .github/workflows/mobile-ios_build.yml:92]: did we settle on these being macos-14 too (not xlarge)?

File: .github/workflows/mobile-ios_build.yml [hunk 3]
```diff
@@ -135,7 +135,7 @@ jobs:
...
```
Comment (abeyad) [inline .github/workflows/mobile-ios_build.yml:138]: same comment above

File: .github/workflows/mobile-ios_tests.yml [hunk 4]
```diff
@@ -50,7 +50,7 @@ jobs:
       docker-ipv6: false
       request: ${{ needs.load.outputs.request }}
       # revert this to non-large once updated
-      runs-on: macos-12-large
+      runs-on: macos-14-xlarge
```
Comment (abeyad) [inline .github/workflows/mobile-ios_tests.yml:53]: same comment above

**Review by abeyad — APPROVED** [review]

**Pull request merged**